test-js/unit/pentaho/type/configurationService.Spec.js

```diff
@@ -0,0 +1,73 @@
+/*!
+ * Copyright 2010 - 2015 Pentaho Corporation.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Must be executed out of strict scope
+var __global__ = this;
+
+define(['pentaho/type/config/ConfigurationService'], function(ConfigurationService) {
+
+  /* global describe:false, it:false, expect:false, beforeEach:false */
+
+  describe("pentaho.type.configurationService -", function() {
+
+    it("returns the instance of the configured service", function() {
+
+      return require.using(["pentaho/type/configurationService"], function(localRequire) {
+        localRequire.config({
+          config: {
+            "pentaho/service": {
+              "testModule": "pentaho.type.IConfigurationService"
+            }
+          }
+        });
+
+        localRequire.define("testModule", "testModuleValue");
+      }, function(configurationService) {
+        expect(configurationService).toBe("testModuleValue");
+      });
+    });
+
+    it("returns a single instance of the configured service even when more than one service is available", function() {
+
+      return require.using(["pentaho/type/configurationService"], function(localRequire) {
+        localRequire.config({
+          config: {
+            "pentaho/service": {
+              "testModule3": "pentaho.type.IConfigurationService",
+              "testModule1": "pentaho.type.IConfigurationService",
+              "testModule2": "pentaho.type.IConfigurationService"
+            }
+          }
+        });
+
+        localRequire.define("testModule1", "testModuleValue1");
+        localRequire.define("testModule2", "testModuleValue2");
+        localRequire.define("testModule3", "testModuleValue3");
+      }, function(configurationService) {
+        expect(["testModuleValue1", "testModuleValue2", "testModuleValue3"]).toContain(configurationService);
+      });
+    });
+
+    it("returns by default a instance of ConfigurationService when no service is configured", function() {
+
+      return require.using(["pentaho/type/configurationService"], function(configurationService) {
```

Comment:
Don't think this test can pass as the top-level ConfigurationService is a class loaded in a different RequireJS context. No?